Content.Shared/Buckle/SharedBuckleSystem.Buckle.cs

```diff
@@ -257,6 +258,9 @@ private bool CanBuckle(EntityUid buckleUid,
         if (!_container.IsInSameOrNoContainer((buckleUid, null, null), (strapUid, null, null)))
             return false;
 
+        if (TryComp<PullableComponent>(strapUid, out var pullable) && pullable.Puller == buckleUid)
+            _pullingSystem.TryStopPull(strapUid, pullable);
+
```

Comment:
This really isn't the proper place for this code. "Can" methods that check if behavior can occur should not contain actual visible behavior within them. It would make sense to either cancel the buckle if you are pulling it, _or_ move the `TryStopPull` call into the `Buckle()` or `SetBuckledTo()` logic

Reply:
I don't want to change game logic by canceling the buckle. I moved it to Buckle